### PR TITLE
chore: add ingest db-helpers

### PR DESCRIPTION
**Pull request by mantagen**
## Description

- updates to `prisma.schema` including
  - adding 'multiSchema' with schema 'ingest' in addition to the default 'public'
  - adding all the required tables to the 'ingest' schema 
- migration file for above changes
- adds 'db-helpers' directory as a with various database helper functions

### CODE REVIEW

File: packages/db/prisma/schema.prisma [hunk 1]
```diff
@@ -276,26 +292,28 @@ model LessonPlan {
...
-  subject      Subject?                     @relation(fields: [subjectId], references: [id], onDelete: Cascade)
-  keyStage     KeyStage?                    @relation(fields: [keyStageId], references: [id], onDelete: Cascade)
+  createdAt    DateTime                    @default(now()) @map("created_at")
+  updatedAt    DateTime                    @updatedAt @map("updated_at")
+  lessonPlan   LessonPlan                  @relation(fields: [lessonPlanId], references: [id], onDelete: Cascade)
+  subject      Subject?                    @relation(fields: [subjectId], references: [id], onDelete: Cascade)
+  keyStage     KeyStage?                   @relation(fields: [keyStageId], references: [id], onDelete: Cascade)
   embedding    Unsupported("vector(256)")?
-  status       LessonPlanPartStatus         @default(PENDING)
+  status       LessonPlanPartStatus        @default(PENDING)
```
Comment (mantagen) [inline packages/db/prisma/schema.prisma:312]: This is just formatting!

File: packages/db/prisma/schema.prisma [hunk 2]
```diff
@@ -843,60 +904,209 @@ enum ModerationType {
...
+}
+
+model IngestLesson {
+  id           String   @id @default(cuid())
+  oakLessonId  Int      @map("oak_lesson_id")
+  ingestId     String   @map("ingest_id")
+  data         Json     @map("data") @db.JsonB
+  dataHash     String   @map("data_hash")
+  step         String
+  stepStatus   String   @map("step_status")
```
Comment (mantagen) [inline packages/db/prisma/schema.prisma:1010]: Currently got `step` and `stepStatus` as text (i.e. not enums) with the typesafety coming from ts helpers, but I'm thinking of using enums instead.

File: packages/db/prisma/schema.prisma [hunk 3]
```diff
@@ -843,60 +904,209 @@ enum ModerationType {
...
+}
+
+model IngestOpenAiBatch {
+  id            String    @id @default(cuid())
+  ingestId      String    @map("ingest_id")
+  inputFilePath String    @map("input_file_path")
+  outputFileId  String?   @map("output_file_id")
+  errorFileId   String?   @map("error_file_id")
+  openaiBatchId String    @map("openai_batch_id")
+  status        String
```
Comment (mantagen) [inline packages/db/prisma/schema.prisma:1087]: Also text, but probably should be enum

File: packages/db/prisma/schema.prisma [hunk 4]
```diff
@@ -843,60 +904,209 @@ enum ModerationType {
...
+  errorMessage  String?   @map("error_message")
+  batchType     String    @map("batch_type")
+
+  ingest Ingest @relation(fields: [ingestId], references: [id])
+
+  @@map("ingest_openai_batch")
+  @@schema("ingest")
+}
+
+model IngestError {
```
Comment (mantagen) [inline packages/db/prisma/schema.prisma:1099]: This needs a bit of work. Planning on adding `errorDetails` or something

File: packages/ingest/src/db-helpers/step.ts [hunk 5]
```diff
@@ -0,0 +1,30 @@
+import { IngestError } from "../IngestError";
+
+export const STEP = [
```
Comment (mantagen) [inline packages/ingest/src/db-helpers/step.ts:3]: This is the order that an ingest happens in.
What stops it from being completely linear is that there are two batch processes for which we have to poll for status from OpenAI.

File: packages/ingest/src/db-helpers/createCaptionsRecord.ts [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+      },
+    });
+    await prisma.ingestLesson.update({
+      where: {
+        id: lessonId,
+      },
+      data: {
+        captionsId: captionsRecord.id,
+      },
+    });
```
[the remark this answers is not in the document]
Comment (mantagen) [reply packages/ingest/src/db-helpers/createCaptionsRecord.ts:33]: Huh, this gives a type error. I was thinking you shouldn't have to do two mutations like this. Will take a look

[the remark this answers is not in the document]
Comment (mantagen) [reply packages/ingest/src/db-helpers/createCaptionsRecord.ts:33]: Have removed the connecting insert ... **should** _just work_.

[the remark this answers is not in the document]
Comment (codeincontext) [reply packages/ingest/src/db-helpers/createCaptionsRecord.ts:33]: Yeah sorry, I'm not 100% sure what the correct syntax is, but it's something along those lines

**Review by codeincontext — APPROVED** [review]

**Pull request merged**